manim/mobject/number_line.py

```diff
@@ -68,6 +68,29 @@ class NumberLine(Line):
     kwargs : Any
         Additional arguments to be passed to :class:`~.Line`.
 
+    Examples
+    --------
+    .. manim:: NumLineExample
+        :save_last_frame:
+        class NumLineExample(Scene):
+            def construct(self):
+                l0 = NumberLine(x_range=[-10, 10, 2], length=10, color=BLUE, include_numbers=True, label_direction=UP)
+                l0.shift(2 * UP)
+
+                l1 = NumberLine(x_range=[-10, 10, 2], unit_size=0.5, numbers_with_elongated_ticks=[-2, 4], include_numbers=True, number_scale_value=0.5)
+                l1.shift(UP)
+                num6 = l1.get_number_mobject(6).set_color(RED)
+
+                l2 = NumberLine(
+                    x_range=[-2.5, 2.5 + 0.5, 0.5], length=12, decimal_number_config={"num_decimal_places": 2}, unit_size=1.25, include_numbers=True
```

Comment:
`length` technically overrides `unit_size` when they're used together. So you should probably chose only one for this line.